app/controllers/proxy_rule_shared_controller.rb

```diff
@@ -12,7 +12,10 @@ def self.included(base)
   end
 
   def index
-    @proxy_rules = proxy_rules.order_by(params[:sort], params[:direction])
+    @search = ThreeScale::Search.new(params[:search])
+    query   = ProxyRuleQuery.new(owner_type: params[:owner_type], owner_id: owner_id)
+    @proxy_rules = query.search_for(@search['query'], proxy_rules)
+                              .order_by(params[:sort], params[:direction])
                               .includes(:metric)
```

Comment:
Can those order and includes be in the query object?

Reply:
yeah, we can do it